gcp_deepvariant_runner.py

```diff
@@ -321,6 +395,83 @@ def _get_gcs_relative_path(gcs_path):
   return urlparse.urlparse(gcs_path).path.strip('/')
 
 
+def _get_bam_category(pipeline_args):
+  """Returns the category that input BAM files belongs to."""
+  bam_size = _get_gcs_object_size(pipeline_args.bam)
+  if bam_size == 0:
+    logging.error('Size of input bam file is 0.')
+
+  is_wes = pipeline_args.model.find(_WES_MODEL_IDENTIFIER) != -1
+  is_wgs = pipeline_args.model.find(_WGS_MODEL_IDENTIFIER) != -1
+
+  if is_wes:
+    if bam_size < _WES_LARGE_THR:
+      return BamCategories.WES_SMALL
+    else:
+      return BamCategories.WES_LARGE
+
+  if is_wgs:
+    if bam_size < _WGS_SMALL_THR:
+      return BamCategories.WGS_SMALL
+    elif bam_size > _WGS_LARGE_THR:
+      return BamCategories.WGS_LARGE
+    else:
+      return BamCategories.WGS_MEDIUM
+
+
+def _set_args_using_optimized_default_flags(default_flags, pipeline_args):
```

Comment:
s/_set_args_using_optimized_default_flags/_set_args_based_on_bam_category
or something easier to grasp.

Also, you don't need to pass _DEFAULT_FLAGS as it's global, just pass bam_category

Reply:
Done.